### PR TITLE
Update Getting Started Docs

### PR DESCRIPTION
**Pull request by NickHeiner**
* Change the "getting started" page to target users who are using this in their own projects, rather than adding examples in the monorepo
* Flesh out the contributing docs
* Add links to the [template repo](https://github.com/fixie-ai/ai-jsx-template/tree/main)
* General docs cleanup
* Add observability docs

If you have CR comments and I've already merged the PR, I'll address them in a follow-up.

### CODE REVIEW

File: packages/docs/docs/guides/observability.md
Comment (NickHeiner) [inline packages/docs/docs/guides/observability.md:1]: @petersalas can you scan this and confirm it's correct?

File: packages/docs/docs/getting-started.md
Comment (NickHeiner) [inline packages/docs/docs/getting-started.md:1]: @larryatfixie, after we merge this PR, you can try this doc and lmk if you're able to create a new project from it.

File: packages/docs/docs/contributing/working-in-the-repo.md
Comment (NickHeiner) [inline packages/docs/docs/contributing/working-in-the-repo.md:1]: This is not new content. It's moved from `contributing.md`.

File: packages/docs/docs/contributing/index.md
Comment (NickHeiner) [inline packages/docs/docs/contributing/index.md:1]: @mdwelsh/@zkoch lmk if you have thoughts on this. I won't wait for y'all to take a look before merging this, but I'll address and comments in a follow-up PR / you can just edit yourself if you prefer.

**Review by mdwelsh — APPROVED** [review]
Thanks for doing this!

**Review by farzadab — APPROVED** [review]
LGTM, thanks Nick!

We might want an "installation" doc as well which the "getting-started" points to later on.

**Pull request merged**